nanvm-lib/src/interface.rs

```diff
@@ -38,8 +38,14 @@ pub trait Function<U: Any<Function = Self>>:
 }
 
 #[derive(Debug, PartialEq)]
-pub enum RuntimeError {
-    TypeError,
+pub enum TypeError<U: Any> {
+    BigIntToNumber(U), // Cannot convert a given BigInt value to a number
+                       // More error variants will be added in the future as needed.
+}
+
+#[derive(Debug, PartialEq)]
+pub enum RuntimeError<U: Any> {
```

Comment:
IMHO, it would be better if we use `Any` for as a run-time error type because FunctionalScript can `throw` anything. For example: https://github.com/functionalscript/functionalscript/blob/main/dev/module.f.ts